lib/src/plugin/plugin.dart

```diff
@@ -8,4 +8,6 @@ abstract class BasePlugin {
 
   FutureOr<void> onBotStart(INyxx nyxx, Logger logger) async {}
   FutureOr<void> onBotStop(INyxx nyxx, Logger logger) async {}
+
+  FutureOr<void> onConnectionChange(INyxx nyxx, Logger logger, int? closeCode, String? closeReason, String? socketError) async {}
```

Comment:
Should this not be another event stream, like [onDisconnect](https://pub.dev/documentation/nyxx/latest/nyxx/IWebsocketEventController/onDisconnect.html)? This would allow users to subscribe to changes without needing to resort to a plugin.

Plugin users can still use the event by subscribing to it in `onBotStart` or `onRegister`.

Reply:
we have already events for disconnects or other stuff I dont think we need another stream for agregating unrelated events (in context to code) so this seems like perfect candidate for a plugin.